### ISSUE TITLE
Listbox label / aria-input-field-name issue

### ISSUE THREAD

**Issue by laurentupa** [dequelabs/axe-core#2998]
Product: axe Extension

This [codepen](https://codepen.io/laurentupa/pen/NWpeWvB) shows the issue I am seeing.

Expectation: Expect that the rule is not applicable in this situation. The unlabeled element with role="listbox" should not fail the aria-input-field-name rule.

Actual: The unlabeled element with role="listbox" fails the aria-input-field-name rule. 

Motivation: Upon the [WAI-ARIA Authoring Practices for listbox](https://www.w3.org/TR/wai-aria-practices-1.1/#listbox_roles_states_props), I came across this rule, which states: 
> If the listbox is not part of another widget, then it has a visible label referenced by aria-labelledby on the element with role listbox.

Because of that statement, I would assume that the listbox should not require an aria-label(led-by) attribute. 

<pre><code>
axe-core version: 4.2.0

Browser and Assistive Technology versions

For Tooling issues:
- Node version: 15.8.0
- Browser: Chrome
</code></pre>

**Comment by straker** [comment]
Interesting, I don't see an `aria-input-field-name` failure for that example. I only see `landmark-one-main`, `nested-interactive`, `page-has-heading-one`, and `region` failures.

![image](https://user-images.githubusercontent.com/2433219/122093524-61080180-cdc8-11eb-872c-2d8f2dfefb01.png)

**Comment by laurentupa** [comment]
I should have mentioned this, but you have to expand the dropdown in order for the role="listbox" to render for the first time. 

![image](https://user-images.githubusercontent.com/52177680/122117293-4e9fbf00-cdec-11eb-894f-ef340208ea3b.png)

**Comment by straker** [comment]
Thanks for the clarification, I can reproduce it now and I'll confirm with @WilcoFiers on what he thinks.

**Comment by craigkovatch** [comment]
@straker we are hitting this one too. Is there a scoped way to disable these particular favors you'd suggest? Any idea on what timeline we might see this fixed?

**Comment by straker** [comment]
My apologies, I never responded back to the original poster. So an element with the [Listbox role](https://www.w3.org/TR/wai-aria-1.2/#listbox) is required to have an accessible name. So having one without an accessible name is indeed a violation. The authoring practices documentation is merely stating that unless the listbox is part of another widget, the listbox needs a _visual_ label and should reference that label using `aria-labelledby`, otherwise it should use `aria-label` to give the listbox an accessible name.

**Comment by craigkovatch** [comment]
@straker Thanks for the quick response! I can certainly understand how we’ve arrived at different interpretations here, and how yours probably makes more sense given the current verbiage. However, I still don’t think this makes any sense.

For a combobox pattern, the list is part of a composite widget and isn’t independently focusable nor navigable. Focus remains on the associated textfield (which has role `combobox`), which *does* require a label. So to me it seems duplicative at best and confusing/misleading at worst to also require a label for the listbox.

I opened https://github.com/w3c/aria-practices/issues/2290 to ask for clarification on this.

Would be interested to hear your thoughts. If we don’t agree, could we please leave this issue open until the ARIA WG can respond on the issue I opened?

**Comment by straker** [comment]
@craigkovatch Going to close this issue as it looks like it's been resolved on the APG / ARIA side of things with no change to the spec (meaning an accessible name is required). Thanks for working on it with all the involved orgs.
